File: BankOOp.py
from abc import ABC, abstractmethod


class bankaccount(ABC):
    def __init__(self, account_number, owner_name, balance):
        self.__account_number = account_number  
        self.__owner_name = owner_name
        self.__balance = balance


    def deposit(self, amount):
        if amount > 0:
            self.__balance += amount
            print(f"New Balance: {self.__balance}")
        else:
            print("Invalid deposit amount.")

    @abstractmethod
    def withdraw(self, amount):
        pass

    # getter method for displaying account info
    def get_account_info(self):
        return {
            "Account Number": self.__account_number,
            "Account Holder": self.__owner_name,
            "Current Balance": self.__balance
        }

    # getter method for displaying balance as __balance in encapuslated we can't use it directly in subclasses
    def get_balance(self):
        return self.__balance

class savingaccount(bankaccount):
    def __init__(self, account_number, owner_name, balance):
        super().__init__(account_number, owner_name, balance)

    def withdraw(self, amount):
        if amount > 0 and amount <= self.get_balance():  # getter method is used here
            self._bankaccount__balance -= amount  
            print(f"New Balance: {self.get_balance()}")
        else:
            print("Invalid deposit amount.")

account = savingaccount("123456", "Punam Adhikari", 1000)
account.deposit(500) 
account.withdraw(300)  
print(account.get_account_info())  

File: BankOOP.java
abstract class bankaccount {
    private String account_number;
    private String owner_name;
    private double balance;


    public bankaccount(String account_number, String owner_name, double balance) {
        this.account_number = account_number;
        this.owner_name = owner_name;
        this.balance = balance;
    }


    public void deposit(double amount) {
        if (amount > 0) {
            balance += amount;
            System.out.println("new balance after deposit: " + balance);
        } else {
            System.out.println("Invalid deposit amount.");
        }
    }

    public abstract void withdraw(double amount);

    public void getaccountinfo() {
        System.out.println("Account Number: " + account_number);
        System.out.println("Owner name: " + owner_name);
        System.out.println("Balance: " + balance);
    }


    public double getbalance() {
        return balance;
    }
}

// inheritance
class savingaccount extends bankaccount {
    public savingaccount(String account_number, String owner_name, double balance) {
        super(account_number, owner_name, balance);
    }

    @Override
    public void withdraw(double amount) {
        if (amount > 0 && amount <= getbalance()) {
            double newbalance = getbalance() - amount;
            System.out.println("new balance after withdrawal: " + newbalance);
        } else {
            System.out.println("Invalid amount.");
        }
    }
}


class banksystem {
    public static void main(String[] args) {

        savingaccount account = new savingaccount("123456", "Punam Adhikari", 1000);

        account.deposit(500);
        account.withdraw(400);
        account.getaccountinfo();
    }
}